#28 SessionService

fmSand opened the pull request:
- "Starter" service - that will change and adapt with more testing, commands and jobs.
- Some methods return `null` for expected failures

**Potential future methods to add:**

 get todays sessions
 get sessions from specific period
 get total session/worked time
 list recent session for user
 user stats
 find idle sessions
 auto close/end sessions

fmSand committed:
feat(services): add SessionService for time tracking

fmSand committed:
chore: jsdocs

fmSand committed:
feat: add getOneActiveByDiscordId to SessionService

Anclagen reviewed (approved):
Looks good overall. Most of the JSDoc feels redundant since the methods are self explanatory, though I get some might prefer having it. normalizeActivity is a bit overly defensive, ideally that validation happens before hitting the DB, but layered safeguards aren’t a bad thing.

fmSand commented:
@Anclagen Tbh, i dont love the jsdocs here while working directly in the file as it feels cluttered. The reason I added them was for development, in case you, or Denis (who's new to JS) might use the methods and you'd get that hover information thingy in vscode.

**normalizeActivity** I forgot to remove in the 'simplify' process (you should see what else I removed lol)!  I can do it later or next time im working on the file. Same with the jsdocs
